api/src/certification/results/domain/models/v3/Certificate.js

```diff
@@ -39,11 +40,15 @@ export class V3Certificate {
     this.deliveredAt = deliveredAt;
     this.certificationCenter = certificationCenter;
     this.pixScore = pixScore;
-    this.globalLevel = new GlobalCertificationLevel({ score: pixScore });
+    this.globalLevel = this.isPreBeginnerLevel ? null : new GlobalCertificationLevel({ score: pixScore });
     this.verificationCode = verificationCode;
     this.maxReachableScore = MAX_REACHABLE_SCORE;
-    this.resultCompetenceTree = resultCompetenceTree;
+    this.resultCompetenceTree = this.isPreBeginnerLevel ? null : resultCompetenceTree;
     this.algorithmEngineVersion = algorithmEngineVersion;
     this.certificationDate = certificationDate;
   }
+
+  get isPreBeginnerLevel() {
+    return new GlobalCertificationLevel({ score: this.pixScore }).meshLevel === LEVELS.preBeginner;
+  }
```

Comment:
Je suis embete par ce getter, car il créé de la logique masquée dans le constructeur et vient réaliser une double instanciation de `new GlobalCertificationLevel`.

Je propose plutôt d'avoir un `get resultCompetenceTree()` qui ressemble à cela et qui permet de faire confiance à notre `this.globalLevel`.

```js
// passer  this.resultCompetenceTree en privé, cad #resultCompetenceTree
// instancier la ligne 46 en mettant uniquement this.#resultCompetenceTree= resultCompetenceTree


get resultCompetenceTree() {
 if(this.globalLevel.meshLevel === LEVELS.preBeginner) {
   return null;
  }

  return this.#resultCompetenceTree;
}
```

Reply:
Autre possibilité : de-responsabiliser le contructeur de créer le `new GlobalCertificationLevel` permettant alors de faire ligne 43 et 46 `this.globalLevel?.meshLevel === LEVELS.preBeginner`

Cela veut dire déporter la logique de new dans tous les repositories qui instancient du `new Certificate` (j'aurais donc tendance à partir sur `get resultCompetenceTree()` proposé ci-dessus.

Reply:
J'ai ajouté un nouveau commit avec une autre solution (que je trouve plus simple), je te laisse faire un retour.